### ISSUE TITLE
FIX S2583: False positive with ref parameter

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#3955]
### Description

Rule S2583 raises a false positive with ref parameters.

Source: https://community.sonarsource.com/t/sonar-is-not-analyzing-the-reference-objects/35834

### Repro steps

```
public void TestExample(ref bool stop)
{
    while (!stop)
    {
        while (true)
        {
            if (stop)
            {
                break;
            }
        }
    }
}
```

### Expected behavior

S2583 should not raise an issue for this case.

### Actual behavior

False positive is reported.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins version: 8.16

**Comment by andrei-epure-sonarsource** [comment]
This seems to be a duplicate of #3353 - or am I missing something?

**Comment by csaba-sagi-sonarsource** [comment]
This issue is a duplicate of [Fix S2589 FP: ref parameters in multithreading](https://github.com/SonarSource/sonar-dotnet/issues/3353)
